retroshare-gui/src/gui/RetroShareLink.cpp

```diff
@@ -730,10 +733,11 @@ void RetroShareLink::check()
 			if(!checkSSLId(_SSLid))
 				_valid = false;			// no break! We also test file stuff below.
 			/* fallthrough */
+			[[clang::fallthrough]];
 		case TYPE_FILE_TREE:
 			/* fallthrough */
 		case TYPE_FILE:
-			if(_size > (((uint64_t)1)<<40))	// 1TB. Who has such large files?
+			if(_size > ((static_cast<uint64_t>(1))<<40))	// 1TB. Who has such large files?
```

Comment:
Reading this code I think RS should not have this kind of limit on file size, as an example think of a big disk image shared trought retroshare, taht could be bigger then 1TB of course, or large databases of LHC experimental data, @csoler what you think of it?

Reply:
I like these limits because they detect a lot of problems. You can end up with arbitrary large files when e.g. deserializing some corrupted data. A very large size can also be the result of some kind of attack with a crafted RSLink. It's probably good to keep that kind of check. Of course the limit can be extended a bit if you want, but 1TB seems reasonnable to me.